akka-typed/src/main/scala/akka/typed/scaladsl/adapter/package.scala

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed
+package scaladsl
+
+import akka.annotation.InternalApi
+import akka.typed.internal.adapter._
+
+/**
+ * Scala API: Adapters between typed and untyped actors and actor systems.
+ * The underlying `ActorSystem` is the untyped [[akka.actor.ActorSystem]]
+ * which runs Akka Typed [[akka.typed.Behavior]] on an emulation layer. In this
+ * system typed and untyped actors can coexist.
+ *
+ * Use these adapters with `import akka.typed.scaladsl.adapter._`.
+ *
+ * Implicit extension methods are added to untyped and typed `ActorSystem`,
+ * `ActorContext`. Such methods make it possible to create typed child actor
+ * from untyped parent actor, and the opposite untyped child from typed parent.
+ * `watch` is also supported in both directions.
+ *
+ * There is an implicit conversion from untyped [[akka.actor.ActorRef]] to
+ * typed [[akka.typed.ActorRef]].
+ *
+ * There are also converters (`toTyped`, `toUntyped`) from typed
+ * [[akka.typed.ActorRef]] to untyped [[akka.actor.ActorRef]], and between untyped
+ * [[akka.actor.ActorSystem]] and typed [[akka.typed.ActorSystem]].
+ */
+package object adapter {
+
+  import language.implicitConversions
+
+  /**
+   * Extension methods added to [[akka.actor.ActorSystem]].
+   */
+  implicit class UntypedActorSystemOps(val sys: akka.actor.ActorSystem) extends AnyVal {
+    def spawnAnonymous[T](behavior: Behavior[T], deployment: DeploymentConfig = EmptyDeploymentConfig): ActorRef[T] =
+      ActorRefAdapter(sys.actorOf(PropsAdapter(Behavior.validateAsInitial(behavior), deployment)))
+    def spawn[T](behavior: Behavior[T], name: String, deployment: DeploymentConfig = EmptyDeploymentConfig): ActorRef[T] =
+      ActorRefAdapter(sys.actorOf(PropsAdapter(Behavior.validateAsInitial(behavior), deployment), name))
+
+    def toTyped: ActorSystem[Nothing] = ActorSystemAdapter(sys)
```

Comment:
I don't think that is exactly the same. If you have an `untyped.ActorRef` that can receive any type of message, so `typed.ActorRef[T]` for any `T` you like should be fine.

In contrast, with `ActorSystem[T]` the `T` defines what messages the user guardian can receive. The user guardian can either be undefined (not defined by user) and then no messages should be allowed, i.e. `Nothing`, or it is a custom user guardian (see `typed.ActorSystem.adapted`) but here we have no clue of what message type that would have, so `Nothing` is the safe choice. If you care about the custom user guardian you should keep around the original `typed.ActorSystem[T]` reference instead.